src/models/AdminTask/WhereClauseBuilder/TargetWhereClauseBuilder/ClauseBuilder.ts

```diff
@@ -0,0 +1,53 @@
+import { Secretary } from "$models/Admin";
+import { ApplicantType } from "$models/Applicant";
+import { Applicant, ApplicantCareer, Offer } from "$models";
+import { AdminTaskType } from "$models/AdminTask/Model";
+
+export const TargetWhereClauseBuilder = {
+  build: (variables: ITargetWhereClauseBuilder) =>
+    [
+      TargetWhereClauseBuilder.getOfferTargetWhereClause(variables),
+      TargetWhereClauseBuilder.getApplicantTypeWhereClause(variables)
+    ]
+      .filter(clause => !!clause)
+      .join(" AND "),
+  getOfferTargetWhereClause: ({ secretary, adminTaskTypes }: ITargetWhereClauseBuilder) => {
+    if (!adminTaskTypes.includes(AdminTaskType.Offer)) return;
+
+    let targetApplicantType: ApplicantType;
+    if (secretary === Secretary.graduados) {
+      targetApplicantType = ApplicantType.graduate;
+    } else {
+      targetApplicantType = ApplicantType.student;
+    }
+
+    return `
+      (
+        "AdminTask"."targetApplicantType" = '${ApplicantType.both}' 
+        OR "AdminTask"."targetApplicantType" = '${targetApplicantType}'
+        OR "AdminTask"."tableNameColumn" != '${Offer.tableName}'
+      )
+    `;
+  },
+  getApplicantTypeWhereClause: ({ secretary, adminTaskTypes }: ITargetWhereClauseBuilder) => {
+    if (!adminTaskTypes.includes(AdminTaskType.Applicant)) return;
+    const isGraduate = secretary === Secretary.graduados;
+    const notOperator = isGraduate ? "" : "NOT";
+
+    return `
+      (
+        "AdminTask"."tableNameColumn" != '${Applicant.tableName}'
+        OR ${notOperator} EXISTS(
```

Comment:
inlinearia este "not operator" porque los nombres de variables son mas confusos, sobre todo isGraduate que me hizo pensar por un momento que era un applicant y no un admin el que llamaba esto